.circleci/config.yml

```diff
@@ -80,10 +80,87 @@ jobs:
 
       # Send notification to discord about the status of the workflow.
       - discord/status
+  build:
+    docker:
+      - image: circleci/openjdk:8-jdk
+    working_directory: ~/repo
+
+    environment:
+      # Customize the JVM maximum heap limit
+      JVM_OPTS: -Xmx3200m
+      TERM: dumb
+    steps:
+      - checkout
+
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "build.gradle" }}
+            # fallback to using the latest cache if no exact match is found
+            - v1-dependencies-
+
+      - run: gradle dependencies
+
+      - save_cache:
+          paths:
+            - ~/.gradle
+          key: v1-dependencies-{{ checksum "build.gradle" }}
+
+      # Build JAR file
+      - run: gradle build
+
+      - run:
+          name: "Move JAR to ./artifacts"
+          command: |
+            mkdir -p artifacts
+            cp ./build/libs/*.jar ./artifacts/
+
+      - run:
+          name: "Save tag number for deployment phase"
+          command: |
+            TAG_STR=$(git describe --all $(git rev-parse HEAD))
+            TAG=$(echo $TAG_STR | perl -ne 'print "$1" if /tags\/(v\d+.\d+.\d+)/')
+            echo -n $TAG > ./artifacts/tag.txt
+
```

Comment:
The only reasonable way to persist the tag number, we only need it in the next job